packages/core/src/components/dialog/multistepDialog.tsx

```diff
@@ -164,28 +164,34 @@ export class MultistepDialog extends AbstractPureComponent2<IMultistepDialogProp
 
     private renderButtons() {
         const { selectedIndex } = this.state;
+        const steps = this.getDialogStepChildren();
         const buttons = [];
+
         if (this.state.selectedIndex > 0) {
+            const backButtonProps = steps[selectedIndex].props.backButtonProps || this.props.backButtonProps;
```

Comment:
nit: avoid implicit boolean type coercion

```suggestion
            const backButtonProps = steps[selectedIndex].props.backButtonProps ?? this.props.backButtonProps;
```